source/lac/trilinos_sparse_matrix.cc

```diff
@@ -1246,15 +1246,15 @@ namespace TrilinosWrappers
 
         Assert(ierr == 0, ExcTrilinosError(ierr));
 
-        int *diag_find =
-          std::find(col_indices, col_indices + num_entries, local_row);
-        int diag_index = static_cast<int>(diag_find - col_indices);
+        const std::ptrdiff_t diag_index =
+          std::find(col_indices, col_indices + num_entries, local_row) -
+          col_indices;
 
         for (TrilinosWrappers::types::int_type j = 0; j < num_entries; ++j)
           if (diag_index != j || new_diag_value == 0)
             values[j] = 0.;
 
-        if (diag_find && std::fabs(values[diag_index]) == 0.0 &&
+        if (diag_index != num_entries && std::fabs(values[diag_index]) == 0.0 &&
```

Comment:
I am pretty sure that this was a bug: `diag_find` would always be a nonzero pointer (unless `col_indices` was `NULL` and `num_entries` was zero, of course)

Reply:
Yes, that's how it looks like.

Reply:
Too bad we don't seem to have a test for this case...